.github/scripts/generate_ci_workflows.py

```diff
@@ -155,6 +156,7 @@ class CIWorkflow:
     only_run_smoke_tests_on_pull_request: bool = False
     num_test_shards_on_pull_request: int = -1
     distributed_test: bool = True
+    timeout_after: int = 240
```

Comment:
Also added a variable timeout here with a default, if we think this is too long we can change this to be something else later on